synthetic_data/graph_synthetic_data.py

```diff
@@ -0,0 +1,130 @@
+'''Generate a synthetic graph using a profile'''
+from cmath import nan
+import math
+
+import dataprofiler as dp
+import networkx as nx
+import scipy.stats as st
+import numpy as np
+
+class GraphDataGenerator():
+    '''
+    Synthesize graph data from a GraphProfiler or graph profile.
+
+    params:
+    graph_profile
+
+    params type:
+    graph_profile: GraphProfiler object/graph_profile
+
+    return:
+    None
+    '''
+
+    def __init__(self, graph):
+
+        if not isinstance(graph, dp.GraphProfiler) or not isinstance(graph, dict):
+            raise NotImplementedError("Graph Profile must be a GraphProfiler object or a dict")
+
+        if isinstance(graph, nx.Graph):
+            data = dp.Data(graph)
+            profiler = dp.GraphProfiler(data)
+            self.profile = profiler.report()
+        else:
+            self.profile = graph
+
+        self._num_nodes = self.profile.get('num_nodes')
+        self._num_edges = self.profile.get('num_edges')
+        self._avg_node_degree = self.profile.get("avg_node_degree")
+        self._categorical_attributes = self.profile.get("categorical_attributes")
+        self._continuous_attributes = self.profile.get("continuous_attributes")
+        self._global_max_component_size = self.profile.get("global_max_component_size")
+        self._continuous_distributions = self.profile.get("continuous_distribution")
+        self._categorical_distributions = self.profile.get("categorical_distribution")
+
+    def synthesize(self):
+        """ Synthesize static graph data with edge attributes. """
+        probability = self._avg_node_degree/self._num_nodes
+
+        graph = nx.erdos_renyi_graph(n=self._num_nodes, p=probability)
+
+        for u,v in graph.edges:
+            edge_attributes = dict()
+
+            for continuous_attribute in self._continuous_attributes:
+                if continuous_attribute is not None:
+                    sample = self.sample_continuous(continuous_attribute)
+                    edge_attributes[continuous_attribute] = sample
+                    
+            for categorical_attribute in self._categorical_attributes:
+                if categorical_attribute is not None:
+                    sample = self.sample_categorical(categorical_attribute)
+                    edge_attributes[categorical_attribute] = sample
```

Comment:
we can condense this into one `for` loop. loop over a list of the variables and then in the nested `for` loop, loop over the contents of the respective variable 
```
for attribute in [continuous_attribute, categorical_attribute]:
> for value in attribute:
...
```